octodns/record/__init__.py

```diff
@@ -162,15 +170,20 @@ def validate(cls, name, fqdn, data):
         return reasons
 
     def __init__(self, zone, name, data, source=None):
+        self.zone = zone
+        if name:
+            # internally everything is idna
+            self.name = idna_encode(str(name))
+            # we'll keep a decoded version around for logs and errors
+            self.decoded_name = idna_decode(self.name)
+        else:
+            self.name = self.decoded_name = name
```

Comment:
Store both our encoded `name` and decoded versions of our name. The former will be used by octoDNS for all of it's work including when the providers refer to/use this record. The latter will mostly be used when printing stuff to the log or in exception messages.